File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.cc
#include <iostream>
#include <set>
using namespace std;

int main() {
  int n;
  set<string, greater<string>> company;

  cin >> n;

  for (int i = 0; i < n; i++) {
    string name, command;
    cin >> name >> command;

    if (command == "enter")
      company.insert(name);

    else if (command == "leave")
      company.erase(name);
  }

  for (auto i = company.begin(); i != company.end(); i++)
    cout << *i << "\n";
}

File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.py
import sys

n = int(sys.stdin.readline())
company = set()

for i in range(n):
  command = sys.stdin.readline().split()

  if command[1] == "enter":
    company.add(command[0])

  elif command[1] == "leave":
    company.remove(command[0])

company = sorted(company)

for i in range(len(company)-1, -1, -1):
  sys.stdout.write(company[i] + "\n")